sdk/tests/core/test_az_mqtt5_connection.c

```diff
@@ -241,36 +252,161 @@ AZ_INLINE void test_az_mqtt5_connection_setup(
       AZ_OK);
 }
 
+AZ_INLINE void test_az_mqtt5_connection_idle_connecting(az_mqtt5_connection* test_conn)
+{
+  will_return(__wrap_az_platform_clock_msec, 0);
+  will_return(__wrap_az_platform_clock_msec, 0);
+  int ref_conn_req_prior = ref_conn_req;
+  assert_int_equal(az_mqtt5_connection_open(test_conn), AZ_OK);
+  assert_int_equal(ref_conn_req_prior + 1, ref_conn_req);
+}
+
+AZ_INLINE void test_az_mqtt5_connection_connecting_connected(az_mqtt5* mock_mqtt5)
+{
+  will_return(__wrap_az_platform_clock_msec, 0);
+  int ref_conn_rsp_ok_prior = ref_conn_rsp_ok;
+
+  assert_int_equal(
+      az_mqtt5_inbound_connack(
+          mock_mqtt5,
+          &(az_mqtt5_connack_data){ .connack_reason = AZ_MQTT5_CONNACK_SUCCESS,
+                                    .tls_authentication_error = false }),
+      AZ_OK);
+  // One pass through the policy collection, another through the callback.
+  assert_int_equal(ref_conn_rsp_ok_prior + 2, ref_conn_rsp_ok);
+}
+
+AZ_INLINE void test_az_mqtt5_connection_connected_connecting(az_mqtt5* mock_mqtt5)
```

Comment:
There is a connected_disconnecting below this inline function, this one goes back to connecting because it receives a disconnect packet so it will try to re-connect